htlcswitch/link.go

```diff
@@ -303,13 +311,29 @@ type ChannelLinkConfig struct {
 	// GetAliases is used by the link and switch to fetch the set of
 	// aliases for a given link.
 	GetAliases func(base lnwire.ShortChannelID) []lnwire.ShortChannelID
-}
 
-// shutdownReq contains an error channel that will be used by the channelLink
-// to send an error if shutdown failed. If shutdown succeeded, the channel will
-// be closed.
-type shutdownReq struct {
-	err chan error
+	// DeliveryAddr is the address to use in Shutdown if an upfront script
+	// has not been set.
+	DeliveryAddr lnwire.DeliveryAddress
```

Comment:
I put this here because if there is no upfront shutdown script, the link needs to be able to send an address in `Shutdown`. An alternative would be to just pass a closure from `peer/brontide.go` instead of having the `peer/brontide.go` code generate it and then store it in the link config

Reply:
Agree with above comment, the link has already been created well before we even know what the upfront shutdown addr is. Generally I don't think we should add more logic to the link (which already has a lot of critical operations), and instead we can use the existing goroutine in the peer to manage the shutdown state.